lib/ddtrace/contrib/resque/resque_job.rb

```diff
@@ -32,7 +33,11 @@ def on_failure_shutdown_tracer(*_)
         end
 
         def shutdown_tracer_when_forked!
-          tracer.shutdown! if forked?
+          return unless forked?
+
+          # Shutdown the tracer and then wait for the writer to flush
+          tracer.shutdown!
+          tracer.writer.worker.join if tracer.writer.is_a?(Datadog::Writer)
```

Comment:
Is there a case where we don't want to wait for the `Datadog::Writer` to finish?
I other words, can we move `tracer.writer.worker.join if tracer.writer.is_a?(Datadog::Writer)` into `tracer.shutdown!`?
My concern is that it looks like we are accessing the internals of `tracer` here, and operating on it.

Reply:
Yeah, I had a similar thought as well.

We might need @delner's input on this one as he might know if this is better as a global change to the tracer or if we should try to scope this to just Resque.

Reply:
I agree with Marco's assessment here: this creates coupling with details about the internal workings of the tracer in Resque, which will make it harder to refactor/change. We should try to encapsulate it in that `shutdown!` method, if possible.

Good consideration though worrying about the broader impact this might have; I think I'd need to do some brief experiments with forking to confirm its behavior, but I hypothesize it would be okay to join the worker thread in the tracer instead of here. I would try leveraging the existing `Datadog::Writer#stop`/`Datadog::Workers::AsyncTransport#stop` function which implements this join for the worker.

Reply:
Perfect, that is definitely an easy change to make here, glad we are all on the same page 👍